package br.com.dukes.todolist.user;

import org.springframework.web.bind.annotation.RestController;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.PostMapping;
import at.favre.lib.crypto.bcrypt.BCrypt;

@RestController 
@RequestMapping("/user")
public class UserController {

    // BCryptService bcryptService = BCryptService.getInstance();

    @Autowired 
    private IUserRepository userRepository;

    @PostMapping("/")
    public ResponseEntity create(@RequestBody UserModel userModel) {
        var user =  this.userRepository.findByUsername(userModel.getUsername());

        if(user != null) {
            // Msg de erro
            // Status Code
            return ResponseEntity.status(HttpStatus.BAD_REQUEST).body("Usuário já existe");
        }

       var passwordHashed = BCrypt.withDefaults()
       .hashToString(12,userModel.getPassword().toCharArray());

       userModel.setPassword(passwordHashed);

        var userCreated = this.userRepository.save(userModel);
        return ResponseEntity.status(HttpStatus.OK).body(userCreated);
    }
}
